test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Sockets.Client;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.FunctionalTests
+{
+    public class HubConnectionTests : IDisposable
+    {
+        private TestServer _testServer;
+        public HubConnectionTests()
+        {
+            var webHostBuilder = new WebHostBuilder().
+                ConfigureServices(services =>
+                {
+                    services.AddSignalR();
+                })
+                .Configure(app =>
+                {
+                    app.UseSignalR(routes =>
+                    {
+                        routes.MapHub<TestHub>("/hubs");
+                    });
+                });
+            _testServer = new TestServer(webHostBuilder);
+        }
+
+        [Fact]
+        public async Task CheckFixedMessage()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = _testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
+                {
+                    var result = await connection.Invoke<string>($"{typeof(TestHub).FullName}.HelloWorld");
+
+                    Assert.Equal("Hello World!", result);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task CheckEchoMessage()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = _testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
+                {
+                    var result =  await connection.Invoke<string>($"{typeof(TestHub).FullName}.Echo", "SignalR");
+
+                    Assert.Equal("SignalR", result);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task CheckCallEcho()
```

Comment:
This test is about invoking client invocations. Consider: CanInvokeClientMethodFromServer